src/Inventory.cpp

```diff
@@ -219,6 +222,50 @@ bool cInventory::RemoveOneEquippedItem(void)
 
 
 
+int cInventory::ReplaceOneEquippedItem(const cItem & a_Item, bool a_TryOtherSlots)
+{
+	// Ignore whether there was an item in the slot to remove.
+	RemoveOneEquippedItem();
+
+	auto EquippedItem = GetEquippedItem();
+	if (EquippedItem.IsEmpty())
+	{
+		SetEquippedItem(a_Item);
+		return a_Item.m_ItemCount;
+	}
+
+	// Handle case when equipped item is the same as the replacement item.
```

Comment:
Hm. When would we remove an item and add the same item back?

Reply:
_We_, as the enlightened programmers, would never do such a thing; plugin developers on the other hand... It's an exported function, why not cover all of the edge cases

Reply:
fair enough